external/mqtt/pom.xml

```diff
@@ -51,7 +51,7 @@
     <dependency>
       <groupId>org.eclipse.paho</groupId>
       <artifactId>org.eclipse.paho.client.mqttv3</artifactId>
-      <version>1.0.1</version>
+      <version>1.0.2</version>
```

Comment:
The MQTT artifact was the one which prompted the creation of https://issues.apache.org/jira/browse/SPARK-4628 in the first place. This newer version is published to Maven Central and has no external dependencies.

@pwendell, interestingly enough this change does not cause `dev/deps` checks to fail because `external/` projects aren't tracked there.